imports/mail/SendAgendaMailHandler.js

```diff
@@ -18,8 +19,14 @@ export class SendAgendaMailHandler extends InfoItemsMailHandler {
     }
     
     _sendMail() {
-        super._sendMail(this._getEmailData());
-    }    
+        Meteor.defer(() => {    // we need .defer here, otherwise the setLocale won't be respected.
```

Comment:
Using this defer here is dangerous. Below the super class' `_sendMail()` method is called which will call the `_buildMail()` method of the `TopicItemsMailHandler` class which initializes the `_mailer` property. The problem is that this function will only be executed after the `sendMail()` method of the `TopicItemsMailHandler` class which resets the `_mailer` property to make sure that the `_mailer` will be freshly initialized for the next user. Using `Meteor.defer()` renders this reset useless, every mail will be sent using the same `_mailer` instance. I don't know how critical that is, but @felixble can probably say more.

Unfortunately, I can't test this without a mail server. Didn't we have a MongoDB mailer for test purposes at some point? Is that still around? Then I can test this and maybe find a better solution.

Reply:
As far as I know the MongoDB mailer is used as default for our e2e tests. Please have a look at the e2e settings json file. I think there is the configuration. If I remember correctly you can use the mailer "test" which writes the emails into a special collection.

Concerning the `Meteor.defer()` topic I have to take a deeper look into this.

Reply:
That was what I remembered as well and it's configured just like that. But I couldn't find the collection. Restarting/reconnecting the MongoDB client helped, I can see the mails now. Thanks!

Reply:
To be honest, I have no clue how `Meteor.defer()` helps here with anything. It does, I verified it, but I don't understand how it works. The `i18n.setLocale()` function is called inside the `defer()` so the promise returned by `setLocale()`won't be resolved before the `super._sendMail()` call. If anything, just deferring the `super._sendMail()` call may help (but comes with the issues outlined above).

Btw, `_sendMail()` doesn't actually send the mail, it's `_buildMail()` which is called by `_sendMail()`. So `_sendMail()` kinda builds the mail and `_buildMail()` sends it…